### ISSUE TITLE
Mislabeled docs for htpasswd-user-groups

### ISSUE THREAD

**Issue by fluzzykitten** [oauth2-proxy/oauth2-proxy#2240]
Documentation says to use variable:
`htpasswd-user-group`
https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview

But go-lang is using instead variable:
`htpasswd-user-groups`
https://github.com/oauth2-proxy/oauth2-proxy/blob/13af1b478633c449a99d7ae9051933a76a41d333/pkg/apis/options/options.go#L35

Yeah, that was annoying to find. They should probably match 🤷

**Comment by ianroberts** [comment]
[This is the case for all legacy options](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/overview#config-file) that are arrays - the command line flag name is singular, the `cfg` file option name is plural.  The mess is gradually getting cleared up as more and more things move to the "alpha config" YAML format.

**Comment by fluzzykitten** [comment]
True true, cfg file ingestion is the problem. I tried to pass a single string in the config as `htpasswd_user_group` and it barfed. So I looked into it and found the cfg file was looking for htpasswd_user_groups. I changed it in the cfg after I saw the disconnect. If the command line flag is right correct (`--htpasswd-user-group`) then it would be nice if the interpretation of the cfg could understand `htpassword_user_group` as well.
